environments/local/quickstart.sh

```diff
@@ -0,0 +1,162 @@
+    printf '.'
+    sleep 1
+  done
+
+  curl \
+    --output /dev/null --silent \
+    --header "Content-Type: application/json" \
+    --request POST \
+    --data '{"audience": ["observatorium"], "client_id": "user", "client_secret": "secret", "grant_types": ["client_credentials"], "token_endpoint_auth_method": "client_secret_basic"}' \
+    http://127.0.0.1:4445/clients
+}
+
+minio() {
+  echo "-------------------------------------------"
+  echo "- Deploying MinIO...  -"
+  echo "-------------------------------------------"
+  $KUBECTL create namespace observatorium-minio --dry-run=client -o yaml | $KUBECTL apply -f -
+
+  $KUBECTL apply -f ./manifests/minio-pvc.yaml
+  $KUBECTL apply -f ./manifests/minio-deployment.yaml
+  $KUBECTL apply -f ./manifests/minio-service.yaml
+
+  echo "-------------------------------------------"
+  echo "- Waiting for MinIO to come up...  -"
+  echo "-------------------------------------------"
+  $KUBECTL wait --for=condition=available --timeout=5m -n observatorium-minio deploy/minio
+}
+
+observatorium() {
+  echo "-------------------------------------------"
+  echo "- Deploying Observatorium...  -"
+  echo "-------------------------------------------"
+  $KUBECTL create namespace observatorium --dry-run=client -o yaml | $KUBECTL apply -f -
+  $KUBECTL apply -f ./manifests/
+
+  echo "-------------------------------------------"
+  echo "- Waiting for Observatorium to come up...  -"
+  echo "-------------------------------------------"
+  $KUBECTL wait --for=condition=available --timeout=5m -n observatorium deploy/observatorium-xyz-thanos-query-frontend
+  $KUBECTL wait --for=condition=available --timeout=5m -n observatorium deploy/observatorium-xyz-observatorium-api
+  ($KUBECTL port-forward -n observatorium svc/observatorium-xyz-observatorium-api 8443:8080 &> /dev/null) &
+}
+
+token_refresher() {
+  echo "-------------------------------------------"
+  echo "- Starting Token Refresher proxy...  -"
+  echo "-------------------------------------------"
+  (./tmp/bin/token-refresher \
+    --oidc.issuer-url=http://172.17.0.1:4444/ \
+    --oidc.client-id=user \
+    --oidc.client-secret=secret \
+    --oidc.audience=observatorium \
+    --url=http://127.0.0.1:8443 &> /dev/null) &
+  sleep 1
+}
+
+prometheus() {
+  echo "-------------------------------------------"
+  echo "- Starting Prometheus...  -"
+  echo "-------------------------------------------"
+  (./tmp/bin/prometheus --config.file=./configs/prom.yaml --storage.tsdb.path=tmp/data/ &> /dev/null) &
+}
+
+grafana() {
+  echo "-------------------------------------------"
+  echo "- Starting Grafana using docker...  -"
+  echo "-------------------------------------------"
+  (docker run -p 3000:3000 grafana/grafana:7.3.7 &> /dev/null) &
+  echo "Open http://localhost:3000 in your browser. Add Prometheus datasource with endpoint http://172.17.0.1:8080/api/metrics/v1/test-oidc."
+}
+
+deploy() {
+  hydra
```

Comment:
I can understand the reason behind creating functions for each of them. However, I'd prefer to inline them if they are not used more than once.